src/Transport/AmqpTransport.php

```diff
@@ -4,45 +4,66 @@
 
 namespace Answear\MessengerHeartbeatBundle\Transport;
 
-use Answear\MessengerHeartbeatBundle\Exception\KeepaliveException;
+use Answear\MessengerHeartbeatBundle\Exception\RabbitMQTransportException;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Messenger\Bridge\Amqp\Transport\AmqpTransport as BaseAmqpTransport;
 use Symfony\Component\Messenger\Bridge\Amqp\Transport\Connection;
 use Symfony\Component\Messenger\Envelope;
 use Symfony\Component\Messenger\Exception\TransportException;
-use Symfony\Component\Messenger\Exception\UnrecoverableMessageHandlingException;
 use Symfony\Component\Messenger\Transport\Receiver\KeepaliveReceiverInterface;
 use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
 
 class AmqpTransport extends BaseAmqpTransport implements KeepaliveReceiverInterface
 {
+    private $errorsCount = 0;
+
     public function __construct(
         private Connection $connection,
         SerializerInterface $serializer,
+        private ?LoggerInterface $logger = null,
     ) {
         parent::__construct($this->connection, $serializer);
     }
 
+    public function get(): iterable
+    {
+        try {
+            return parent::get();
+        } catch (TransportException $transportException) {
+            $this->logger?->warning(
+                '[Keepalive] TransportException has been thrown.',
+                ['exception' => $transportException]
+            );
+
+            throw RabbitMQTransportException::createException($transportException);
```

Comment:
to jest problem z prod. Keepalive było ogarnianie poprawnie, ale PRECONDITION dostała wiadomość i to już nie było ogarniane w tym flow.

Wcześniej to było jakoś inaczej, że jedno i drugie wpadało w to samo flow i było git.